src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -791,7 +792,11 @@ final case class ClientApiServiceImpl(
     } yield (clientPurpose, purpose, agreement, eService, descriptor)
 
     for {
-      consumer              <- partyManagementService.getInstitution(client.consumerId)
+      tenant                <- tenantManagementService
+        .getTenant(client.consumerId)
+      selfcareId            <- tenant.selfcareId.toFuture(ClientRetrievalError)
+      selfcareUUID          <- selfcareId.toFutureUUID
```

Comment:
we consider `selfcareId` as `String`, what do you think about doing this adapting in the `partyManagementService.getInstitution`?